src/main/java/cucumber/testdata/DatabaseTesterNewSchemaDesign.java

```diff
@@ -4885,4 +4885,121 @@ public void updateAppearanceForShowCause(String jurorNumber) throws SQLException
+			conn = db.getConnection(env_property);
+		else
+			conn = db.getConnection("demo");
+
+		try {
+
+			pStmt = conn.prepareStatement("delete from juror_mod.holiday");
+			pStmt.executeUpdate();
+
+
+		} catch (SQLException e) {
+			e.printStackTrace();
+			log.error("Message:" + e.getMessage());
+
+		} finally {
+			conn.commit();
+			pStmt.close();
+			conn.close();
+		}
+	}
+
+	public void addBankHolidayForYear2024() throws SQLException {
+		db = new DBConnection();
+
+		String env_property = System.getProperty("env.database");
+
+		if (env_property != null)
+			conn = db.getConnection(env_property);
+		else
+			conn = db.getConnection("demo");
+
+		try {
+
+			pStmt = conn.prepareStatement("INSERT INTO juror_mod.holiday (loc_code, holiday, description, public) VALUES (null,'2024-03-29','Good Friday',true)");
+			pStmt.executeUpdate();
+
+			pStmt = conn.prepareStatement("INSERT INTO juror_mod.holiday (loc_code, holiday, description, public) VALUES (null,'2024-04-01','Easter Monday',true)");
+			pStmt.executeUpdate();
+
+			pStmt = conn.prepareStatement("INSERT INTO juror_mod.holiday (loc_code, holiday, description, public) VALUES (null,'2024-05-06','Early May bank holiday',true)");
+			pStmt.executeUpdate();
+
+			pStmt = conn.prepareStatement("INSERT INTO juror_mod.holiday (loc_code, holiday, description, public) VALUES (null,'2024-05-27','Spring bank holiday',true)");
+			pStmt.executeUpdate();
+
+			pStmt = conn.prepareStatement("INSERT INTO juror_mod.holiday (loc_code, holiday, description, public) VALUES (null,'2024-08-26','Summer bank holiday',true)");
+			pStmt.executeUpdate();
+
+			pStmt = conn.prepareStatement("INSERT INTO juror_mod.holiday (loc_code, holiday, description, public) VALUES (null,'2024-12-25','Christmas Day',true)");
+			pStmt.executeUpdate();
+
+			pStmt = conn.prepareStatement("INSERT INTO juror_mod.holiday (loc_code, holiday, description, public) VALUES (null,'2024-12-26','Boxing Day',true)");
+			pStmt.executeUpdate();
+
+		} catch (SQLException e) {
+			e.printStackTrace();
+			log.error("Message:" + e.getMessage());
+
+		} finally {
+			conn.commit();
+			pStmt.close();
+			conn.close();
+		}
+	}
+	public void addBankHolidayForYear2025() throws SQLException {
+		db = new DBConnection();
+
+		String env_property = System.getProperty("env.database");
+
+		if (env_property != null)
+			conn = db.getConnection(env_property);
+		else
+			conn = db.getConnection("demo");
+
+		try {
+
+			pStmt = conn.prepareStatement("INSERT INTO juror_mod.holiday (loc_code, holiday, description, public) VALUES (null,'2025-01-01','New years Day',true)");
+			pStmt.executeUpdate();
+
+			pStmt = conn.prepareStatement("INSERT INTO juror_mod.holiday (loc_code, holiday, description, public) VALUES (null,'2025-04-18','Good Friday',true)");
+			pStmt.executeUpdate();
+
+			pStmt = conn.prepareStatement("INSERT INTO juror_mod.holiday (loc_code, holiday, description, public) VALUES (null,'2025-04-21','Easter Monday',true)");
```

Comment:
I'm not sure about these, if we insert fixed dates there may be a point in the future where they fail, I think we should dynamically insert less holidays and check those.

Reply:
Add new steps based on the discussion